src/dndbeyond/content-scripts/character.js

```diff
@@ -811,28 +811,13 @@ async function rollItem(force_display = false, force_to_hit_only = false, force_
                 let damage = value.find(".ct-damage__value,.ddbc-damage__value").text();
                 let damage_type = properties["Damage Type"] || "";
                 let versatile_damage = value.find(".ct-item-detail__versatile-damage,.ddbc-item-detail__versatile-damage").text().slice(1, -1);
-                if (damages.length == 0 && character.hasGreatWeaponFighting(2014) &&
-                    properties["Attack Type"] == "Melee" &&
-                    (properties["Properties"].includes("Versatile") || properties["Properties"].includes("Two-Handed"))) {
-                    if (versatile_damage != "") {
-                        versatile_damage = versatile_damage.replace(/[0-9]*d[0-9]+/g, "$&ro<=2");
-                    } else {
-                        damage = damage.replace(/[0-9]*d[0-9]+/g, "$&ro<=2");
-                    }
-                }
-                if (damages.length == 0 && character.hasGreatWeaponFighting(2024) &&
-                    properties["Attack Type"] == "Melee" &&
-                    (properties["Properties"].includes("Versatile") || properties["Properties"].includes("Two-Handed"))) {
-                    if (versatile_damage != "") {
-                        versatile_damage = versatile_damage.replace(/([0-9]*)d([0-9]+)([^\s+-]*)(.*)/g, (match, amount, faces, roll_mods, mods) => {
-                            return new Array(parseInt(amount) || 1).fill(`1d${faces}${roll_mods}min3`).join(" + ") + mods;
-                        });
-                    } else {
-                        damage = damage.replace(/([0-9]*)d([0-9]+)([^\s+-]*)(.*)/g, (match, amount, faces, roll_mods, mods) => {
-                            return new Array(parseInt(amount) || 1).fill(`1d${faces}${roll_mods}min3`).join(" + ") + mods;
-                        });
-                    }
+
```

Comment:
There was a check here for `if (damages.length == 0)` which disappeared in the refactor.
I think it was because this code runs in a loop, so if the weapon has multiple damages (flaming sword?) then it could get applied multiple times, so `if (damages.length == 0)` makes sure it only executes it once.

Reply:
good point yeah that should be the case if there is multiple damage die. things like flamesword the fire doesnt get gwf